.gitignore

```diff
@@ -22,7 +22,11 @@
 test_data
 trie_putandget_failed_test_data_*
 tmp
+lib/erasure
```

Comment:
What is this, is it generated somewhere?

Reply:
Yes, this is where the compiled Rust lib for erasure coding ends up.

Reply:
`lib/erasure` also contains Go code. we need to ignore `lib/erasure/rustlib` instead.
What do you think?

Reply:
I recently saw `lib/erasure` under "untracked files" in the output of `git status` and when I saw it being added here, assumed that it only contains compilation artifacts.

I just wanted to double check and was confused that I couldn't find `lib/erasure` in the repo at all any more. It turns out [Axays PR that adds the Rust lib](https://github.com/ChainSafe/gossamer/pull/3447) wasn't merged to `development` but to `feat/parachain`. (It seems like the two branches have diverged quite a bit. Maybe `feat/parachain` should be rebased on `development`?)

Anyway, given this situation, if we ignore `lib/erasure` on `development`, we don't want to bring that change to `feat/parachain`. If we don't ignore it, we risk accidentally adding the directory on `development`. 🤔

Reply:
> Anyway, given this situation, if we ignore lib/erasure on development, we don't want to bring that change to feat/parachain. If we don't ignore it, we risk accidentally adding the directory on development. 🤔

I'll remove this from this PR then.  We definitely need this for the `feat/parachain` branch.  For context, that's where all the work that is done regarding parachain subsystems, and we have a long lived feature branch that we haven't rebased in a while.  I included it since locally I have been switching between `development` and `feat/parachain` and so I have it in my `.gitignore`.